bindings/node/test/autoEncrypter.test.js

```diff
@@ -0,0 +1,153 @@
+      .callsFake((client, ns, command, callback) => {
+        if (ENABLE_LOG_TEST) {
+          const response = BSON.deserialize(MOCK_MONGOCRYPTD_RESPONSE);
+          response.schemaRequiresEncryption = false;
+
+          ENABLE_LOG_TEST = false; // disable test after run
+          callback(null, BSON.serialize(response));
+          return;
+        }
+
+        callback(null, MOCK_MONGOCRYPTD_RESPONSE);
+      });
+
+    sandbox.stub(StateMachine.prototype, 'fetchKeys').callsFake((client, ns, filter, callback) => {
+      // mock data is already seriaized, our action deals with the result of a cursor
+      const deserializedKey = BSON.deserialize(MOCK_KEYDOCUMENT_RESPONSE);
+      callback(null, [deserializedKey]);
+    });
+  });
+
+  after(() => {
+    sandbox.restore();
+  });
+
+  describe('state machine', function() {
+    it('should decrypt mock data', function(done) {
+      const input = readExtendedJsonToBuffer(`${__dirname}/data/encrypted-document.json`);
+      const client = new MockClient();
+      const mc = new AutoEncrypter(client, {
+        keyVaultNamespace: 'admin.datakeys',
+        logger: () => {},
+        kmsProviders: {
+          aws: { accessKeyId: 'example', secretAccessKey: 'example' },
+          local: { key: Buffer.alloc(96) }
+        }
+      });
+
+      mc.decrypt(input, (err, decrypted) => {
+        if (err) return done(err);
+        expect(decrypted).to.eql({ filter: { find: 'test', ssn: '457-55-5462' } });
+        done();
+      });
+    });
+
+    it('should encrypt mock data', function(done) {
+      const client = new MockClient();
+      const mc = new AutoEncrypter(client, {
+        keyVaultNamespace: 'admin.datakeys',
+        logger: () => {},
+        kmsProviders: {
+          aws: { accessKeyId: 'example', secretAccessKey: 'example' },
+          local: { key: Buffer.alloc(96) }
+        }
+      });
+
+      mc.encrypt('test.test', TEST_COMMAND, (err, encrypted) => {
+        if (err) return done(err);
+        expect(encrypted).to.have.keys('filter', 'find');
+        expect(encrypted.filter).to.have.keys('ssn');
+        done();
+      });
+    });
+  });
+
+  describe('logging', function() {
+    it('should allow registration of a log handler', function(done) {
+      ENABLE_LOG_TEST = true;
+
+      let loggerCalled = false;
+      const logger = (level, message) => {
+        if (loggerCalled) return;
+
+        loggerCalled = true;
+        expect(level).to.be.oneOf([2, 3]);
+        expect(message).to.not.be.empty;
+      };
+
+      const client = new MockClient();
+      const mc = new AutoEncrypter(client, {
+        logger,
+        kmsProviders: {
+          aws: { accessKeyId: 'example', secretAccessKey: 'example' },
+          local: { key: Buffer.alloc(96) }
+        }
+      });
+
+      mc.encrypt('test.test', TEST_COMMAND, (err, encrypted) => {
+        if (err) return done(err);
+        expect(encrypted).to.have.keys('filter', 'find');
```

Comment:
Same with this test.

Reply:
same thing as above. This is just copy-paste so that we could trigger a code path that makes a logging message occur